### PR TITLE
Enable and improve OMR module

### PR DESCRIPTION
**Pull request by lasconic**

### CODE REVIEW

**Comment by lasconic** [comment]
This looks good. I have 3 more things before merging this.

1. Remove the PDFs. It's not a good idea to store these on git. Let's move them somewhere else. Attached to the issue https://musescore.org/node/110306 if necessary.

2. I need to make sure it compiles and run on windows without additional dependencies.

3. Squash everything and use "Fix #110306: Enable and improve OMR module"

File: omr/omr.cpp [hunk 1]
```diff
@@ -139,70 +142,123 @@ int Omr::pagesInDocument() const
...
             _ocr = new Ocr;
       _ocr->init();
 #endif
+      int ID = READ_PDF;
+      int page = 0;
+      bool val;
+      while (ID < ACTION_NUM) {
+            if(ID != INIT_PAGE && ID != SYSTEM_IDENTIFICATION) {
+                  page = 0;
+                  progress->setLabelText(ActionNames.at(ID+1) + " at Page" + QString::number(1));
```
Comment (lasconic) [inline omr/omr.cpp]: This need to be translated and there is a missing space. See also the other line. It should look like 
```
tr("%1 at Page %2").arg(ActionNames.at(ID+1)).arg(1)
```

**Comment by AntonioBL** [comment]
Under Windows, I found that the "include" files for libcurl and libpng are needed. Moreover, HAVE_GMTIME_R and HAVE_RAND_R have to be undef (preprocessor instruction) for grandom.cc and DateInfo.cc (maybe these are not really used by the required Poppler instructions, since MuseScore does not complain when the libcurl and libpng dlls are missing).
Beside these points, the compilation runs fine and the code seems to work as expected.
Tested under Windows 10 Home, gcc 4.9.2, Qt 5.6

**Comment by AntonioBL** [comment]
Here is a screenshot under Windows.
omr_windows

File: thirdparty/poppler/config.h [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+/* #undef HAVE_PREAD64 */
+
+/* Define if you have POSIX threads libraries and header files. */
+#define HAVE_PTHREAD 1
+
+/* Have PTHREAD_PRIO_INHERIT. */
+#define HAVE_PTHREAD_PRIO_INHERIT 1
+
+/* Defines if rand_r is available on your system */
+//#define HAVE_RAND_R 1
```
Comment (liang-chen) [inline thirdparty/poppler/config.h]: undefine HAVE_RAND_R

File: thirdparty/poppler/config.h [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+#define HAVE_FSEEKO 1
+
+/* Define to 1 if you have the `ftell64' function. */
+/* #undef HAVE_FTELL64 */
+
+/* Defines if gettimeofday is available on your system */
+#define HAVE_GETTIMEOFDAY 1
+
+/* Defines if gmtime_r is available on your system */
+//#define HAVE_GMTIME_R 1
```
Comment (liang-chen) [inline thirdparty/poppler/config.h]: undefine HAVE_GMTIME_R

**Comment by liang-chen** [comment]
PDF has already removed from this branch, the results are attached to the issue right now. Squashed latest commits and use "Fix #110306: Enable and improve OMR module". Undefined HAVE_GMTIME_R and HAVE_RAND_R in poppler/config.h.

**Comment by Jojo-Schmitz** [comment]
lots of compiler warnings in poppler code

**Comment by liang-chen** [comment]
Fixed a portion of poppler warnings by removing the name of extra arguments.

**Pull request merged**